ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.java

```diff
@@ -0,0 +1,232 @@
+     */
+    @NotNull
+    List<ContractState> getEncumbranceGroup(@NotNull String tag);
+
+    /**
+     * Gets a map of encumbrance group tags and the associated encumbered {@link ContractState} instances.
+     *
+     * @return Returns map of encumbrance group tags and the associated encumbered {@link ContractState} instances.
+     */
+    @NotNull
+    Map<String, List<ContractState>> getEncumbranceGroups();
+
+    /**
+     * Gets the notary assigned to the current transaction, or null if the notary has not been set.
+     *
+     * @return Returns the notary assigned to the current transaction, or null if the notary has not been set.
+     */
+    @Nullable
+    Party getNotary();
+
+    /**
+     * Sets the specified {@link Party} as a notary to the current {@link UtxoTransactionBuilder}.
+     *
+     * @param notary The {@link Party} to set as a notary to the current {@link UtxoTransactionBuilder}.
+     * @return Returns a new {@link UtxoTransactionBuilder} with the new notary.
+     */
+    @NotNull
+    UtxoTransactionBuilder setNotary(@NotNull Party notary);
+
+    /**
+     * Sets the transaction time window to be valid until the specified {@link Instant}, tending towards negative infinity.
+     *
+     * @param until The {@link Instant} until which the transaction time window is valid.
+     * @return Returns the current {@link UtxoTransactionBuilder} including the specified time window.
+     */
+    @NotNull
+    UtxoTransactionBuilder setTimeWindowUntil(@NotNull Instant until);
+
+    /**
+     * Sets the transaction time window to be valid between the specified {@link Instant} values.
+     *
+     * @param from  The {@link Instant} from which the transaction time window is valid.
+     * @param until The {@link Instant} until which the transaction time window is valid.
+     * @return Returns the current {@link UtxoTransactionBuilder} including the specified time window.
+     */
+    @NotNull
+    UtxoTransactionBuilder setTimeWindowBetween(@NotNull Instant from, @NotNull Instant until);
+
+    /**
+     * Verifies the content of the {@link UtxoTransactionBuilder} and
+     * signs the transaction with any required signatories that belong to the current node.
+     * <p>
+     * Calling this function once consumes the {@link UtxoTransactionBuilder}, so it cannot be used again.
+     * Therefore, if you want to build two transactions you need two builders.
+     *
+     * @return Returns a {@link UtxoSignedTransaction} with signatures for any required signatories that belong to the current node.
+     * @throws IllegalStateException when called a second time on the same object to prevent unintentional duplicate transactions.
+     * @throws TransactionNoAvailableKeysException if none of the required keys are available to sign the transaction.
+     */
+    @NotNull
+    @Suspendable
+    UtxoSignedTransaction toSignedTransaction() throws IllegalStateException, TransactionNoAvailableKeysException;
```

Comment:
Not sure throws clause is correct as both should be runtime exceptions and if not need changing!

Reply:
All `throws` clauses removed.